_tasks/sync-proposal-data.mjs

```diff
@@ -0,0 +1,303 @@
+  const codeBlocks = [];
+  // Some proposal repos use a different branch (e.g., `gh-pages`) for their
+  // README file. Because of this, we need repo metadata to find the raw URL.
+  let rawReadmeUrl;
+  await ghFileImporter
+    .fetchPathMetadata('tc39', prpslId, 'README.md')
+    .then((result) => {
+      rawReadmeUrl = result.data.download_url;
+    });
+  // Get the file and parse out the code samples.
+  await ghFileImporter.importFileFromUrl(rawReadmeUrl).then(async (result) => {
+    const docSourceText = result;
+    await unified()
+      .use(markdown)
+      .use(() => {
+        return (tree) => {
+          visit(tree, null, (node) => {
+            if (
+              node &&
+              node.type === 'code' &&
+              node.lang.match(REGEX_JS_INFO_STRS)
+            ) {
+              codeBlocks.push(he.encode(node.value));
+            }
+          });
+        };
+      })
+      .use(remark2rehype, { allowDangerousHtml: true })
+      .use(raw)
+      .use(htmlStringify)
+      .process(docSourceText);
+  });
+  return codeBlocks.length > 0 ? codeBlocks.pop() : undefined;
+}
+
+/**
+ * @param {string} linkTextLike
+ * @returns {string} The proposal ID.
+ */
+function getPrpslId(linkTextLike) {
+  let prpslUrl = getUrlFromDoc(linkTextLike);
+  if (prpslUrl.endsWith('/')) {
+    prpslUrl = prpslUrl.slice(0, -1); // Ditch trailing slashes & prevent crash.
+  }
+  return prpslUrl
+    .split('/')
+    .pop()
+    .toLowerCase();
+}
+
+/**
+ * Checks if the proposal has a spec using the files listed by the GitHub API.
+ * @param {string} prpslId
+ * @returns {boolean}
+ */
+async function hasSpec(prpslId) {
+  let isFound = false;
+  await ghFileImporter.fetchPathMetadata('tc39', prpslId).then((result) => {
+    result.data.forEach((value) => {
+      if (value.path === 'spec.html') {
+        isFound = true;
+      }
+    });
+  });
+  return isFound;
+}
+
+/**
+ * @param {Uint8Array} buffer
+ * @param {Object} options
+ * @returns {Uint8Array}
+ */
+function json2yaml(buffer, options) {
+  const src = JSON.parse(buffer.toString());
+  const ymlDocument = options.safe
+    ? yaml.safeDump(src, options)
+    : yaml.dump(src, options);
+  return Buffer.from(ymlDocument);
+}
+
+/**
+ * Does it already have a description by reading the yaml from stage_3.yml
+ * @param {string}
+ * @returns {string}
+ */
+async function getDescription(prpslId) {
+  const data = await readFile(pathResolve(DIR_DATA, 'stage3.yml'), 'utf8');
+  let description;
+  yaml.safeLoadAll(data, (doc) => {
+    doc.forEach((value) => {
+      if (ownProperty(value, 'id') && value.id === prpslId) {
+        // Proposal description has already has already been filled out.
+        log.info(
+          `Using pre-existing description for ${curlyQuote(prpslId)}: ${
+            value.description
+          }`
+        );
+        description = value.description;
+      }
+    });
+  });
+  if (!description) {
+    const response = await prompt({
```

Comment:
@ljharb, now that that's taken care of... By any chance would you happen to have any idea why this hangs if the description is missing?

Reply:
No - does it prompt before it hangs? or does it never get to the prompt? If you console.log in a bunch of places, where's the last place that gets run?

Reply:
Well, it halts execution at the right point, but the prompt doesn't show up &mdash; there are no words. You need to press enter or ctrl+c to be able to move forward and only then does it appear. 

I hope this doesn't have to do w/ the fact that I'm using a custom logger.

Reply:
I'm not familiar with enquirer, so that might be the case :-/

Reply:
This combination is supposed to work. Strange indeed. Will have to investigate alternatives.